skimage/filters/_gaussian.py

```diff
@@ -83,23 +84,29 @@ def gaussian(image, sigma=1, output=None, mode='nearest', cval=0,
     >>> from skimage.data import astronaut
     >>> image = astronaut()
     >>> filtered_img = gaussian(image, sigma=1, multichannel=True)
-
     """
-
     spatial_dims = guess_spatial_dimensions(image)
     if spatial_dims is None and multichannel is None:
         msg = ("Images with dimensions (M, N, 3) are interpreted as 2D+RGB "
                "by default. Use `multichannel=False` to interpret as "
                "3D image with last dimension of length 3.")
         warn(RuntimeWarning(msg))
         multichannel = True
+
     if np.any(np.asarray(sigma) < 0.0):
         raise ValueError("Sigma values less than zero are not valid")
+
+    # Handle longstanding SciPy bug with NumPy integers passed as sigma
```

Comment:
If you have a thin area which was white around an otherwise darkened image, using 'nearest' will result in a massive white halo in that area. Or, similarly, a thin black region will make a large regional darkening. Neither of these are per se bugs - you can see why they happen - but they are not intuitive nor what a user will expect when blurring the image.

Gaussian filtering should always be done with `reflect`. I can't think of any instances where this isn't true. 

Multiple tests in the `feature` suite broke when I substituted without changing the default mode, notably in the `blob_*` methods. Changing the edge padding method matters. From a regression standpoint, we're already covered by those tests (this change is the fix). 

Even if you disregard all of that, the fact that we state up front it's a simple wrapper around `ndi.gaussian_filter` _and then change the defaults_ is close to malicious. This needs to be fixed.